spack/scripts/update-metadata.py

```diff
@@ -1,90 +1,100 @@
-                    regex_available_module = re.compile(
-                        cray_name + r"\/(\S*?)(\(default\))?\s+"
-                    )
-                    available_versions = regex_available_module.findall(
-                        cmd.stderr.decode("utf-8")
-                    )
-
-                    for (cray_version, _) in available_versions:
-                        for spack_specs in module_mappings[cray_name]:
-                            if spack_name == "cuda":
-                                cuda_version = cray_version.split(".")
-                                spack_version = f"{cuda_version[0]}.{cuda_version[1]}"
-                            else:
-                                spack_version = cray_version
-
-                            full_spack_name = f"{spack_name}@{spack_version}"
-                            if spack_specs != "":
-                                full_spack_name = f"{full_spack_name}%{spack_specs}"
-                            new_modules[full_spack_name] = f"{cray_name}/{cray_version}"
-
-                    default_versions = [
-                        v for (v, d) in available_versions if d == "(default)"
-                    ]
-                    if spack_name == "cuda":
-                        cuda_version = default_versions[0].split(".")
-                        spack_version = f"{cuda_version[0]}.{cuda_version[1]}"
-                    else:
-                        spack_version = default_versions[0]
-
-                    new_data["packages"][spack_name]["version"] = [spack_version]
-
-                new_data["packages"][spack_name]["modules"] = new_modules
 
+                for cray_version, _ in available_versions:
+                    for spack_specs in module_mappings[cray_name]:
+                        if spack_pkg_name == "cuda":
+                            cuda_version = cray_version.split(".")
+                            spack_version = f"{cuda_version[0]}.{cuda_version[1]}"
+                        else:
+                            spack_version = cray_version
+
+                        full_spack_name = f"{spack_pkg_name}@{spack_version}"
+                        if spack_specs != "":
+                            full_spack_name = f"{full_spack_name}%{spack_specs}"
+                        new_modules[full_spack_name] = f"{cray_name}/{cray_version}"
+
+                default_versions = [
+                    v for (v, d) in available_versions if d == "(default)"
+                ]
+                if spack_pkg_name == "cuda":
+                    cuda_version = default_versions[0].split(".")
+                    spack_version = f"{cuda_version[0]}.{cuda_version[1]}"
+                else:
+                    spack_version = default_versions[0]
+
+                new_data["packages"][spack_pkg_name]["version"] = [spack_version]
+
+            new_data["packages"][spack_pkg_name]["modules"] = new_modules
+
+    with open(args.output_file, "w") as outfile:
```

Comment:
You should check that `output_file` is not the same as `input_file`.

Reply:
Why not allow to overwrite a file?

Reply:
What if you accidentally overwrite the original `packages.yaml` file? But if that's a valid usage scenario, I have not a problem.

Reply:
Right now we are keeping all the versions of the modules in one file, so in theory ` spack/dom/packages.yaml` will be getting bigger and be rewritten.